base/file.jl

```diff
@@ -230,23 +230,27 @@ julia> mkpath("intermediate_dir/actually_a_directory.txt") # creates two directo
 julia> isdir("intermediate_dir/actually_a_directory.txt")
 true
 
+julia> mkpath("my/test/dir/") # returns the original `path`
+"my/test/dir/"
 ```
 """
 function mkpath(path::AbstractString; mode::Integer = 0o777)
-    isdirpath(path) && (path = dirname(path))
```

Comment:
Your patch seems to have effects besides what its stated goal is. That adds extra work for reviewing.

How about just inserting `origpath = path` at the top, and changing the final line to `return origpath` ? Then that's trivial to review.

Reply:
Thanks for your comment!

The patch includes two changes:
1. deleting the first line `isdirpath(path) && (path = dirname(path))`
2. skip `mkdir(path)` if the path is already a dir.

The second change is only optional, and I added it just to improve performance. It does not change the effect.

I checked the history of the first line. It was added by @StefanKarpinski  to fix #4342 (a4957a7dfaf743aa0cae19bedeb2468898767d8e),  which enables `mkpath` using paths such as "x/".
The `mkpath` doesn't return `path` at that time, which was implemented in #27071 and  #29990.

I also carefully checked some cases, and feel confident about the changes.
I agree that the solution is harder to verify than simpler ones, e.g., inserting `origpath = path`.
However, I think it's better because the patch reduces the complexity of the code for future maintenance.

Reply:
I don't find it simpler at all, perhaps because it is mostly uncommented and the function names are not particularly self-explanatory. And your new comments have issues with grammar/spelling :-/